tests/python_package_test/test_dask.py

```diff
@@ -262,6 +262,37 @@ def _unpickle(filepath, serializer):
         raise ValueError(f'Unrecognized serializer type: {serializer}')
 
 
+def _objective_least_squares(y_true, y_pred):
+    grad = y_pred - y_true
+    hess = np.ones(len(y_true))
+    return grad, hess
+
+
+def _objective_logistic_regression(y_true, y_pred):
+    y_pred = 1.0 / (1.0 + np.exp(-y_pred))
+    grad = y_pred - y_true
+    hess = y_pred * (1.0 - y_pred)
+    return grad, hess
+
+
+def _objective_logloss(y_true, y_pred):
+    num_rows = len(y_true)
+    num_class = len(np.unique(y_true))
+    # operate on preds as [num_data, num_classes] matrix
+    y_pred = y_pred.T.reshape(-1, num_class)
```

Comment:
The `y_pred` array is 1D and comes in the form [probs_class1 | probs_class2 | probs_class3] so we have to reshape it with `order='F'`.
```suggestion
    y_pred = y_pred.reshape(-1, num_class, order='F')
```

Reply:
ah!! you're absolutely right! thank you.

```python
import numpy as np

x = np.array([
    0.1, 0.1, 0.2, 0.2, 0.3, 0.3
])

x.T.reshape(-1, 3)

# array([
#    [0.1, 0.1, 0.2],
#    [0.2, 0.3, 0.3]
# ])

x.reshape(-1, 3, order='F')

# array([
#    [0.1, 0.2, 0.3],
#    [0.1, 0.2, 0.3]
# ])
```